### PR TITLE
feat: add Kroma Burgundy upgrade time

### PR DESCRIPTION
**Pull request by Pangssu**
The Kroma Burgundy upgrade activates the governance token. 
This PR includes the implementation of configurations for the Burgundy upgrade.

## Summary by CodeRabbit

- **New Features**
	- Introduced predeploy configuration management for Ethereum smart contracts.
	- Added a new `L2GenesisBurgundyTimeOffset` field to control the activation time of the Kroma Burgundy hard fork.
	- Implemented methods to calculate and set the Kroma Burgundy hard fork activation time.
	- Enhanced network upgrade functionality with the Kroma Burgundy update, including new configuration fields and methods.

- **Refactor**
	- Updated file permissions for enhanced security in writing JWT secret.

- **Chores**
	- Adjusted setup configurations to handle new Burgundy time offset.

### CODE REVIEW

**Comment by kangsorang** [comment]
While not currently used, it would be nice to have the flag code in the Op-e2e descendant for consistency. (setup.go)

**Review by sm-stack — APPROVED** [review]
LGTM

**Review by kangsorang — APPROVED** [review]
LGTM

File: kroma-chain-ops/genesis/genesis.go [hunk 1]
```diff
@@ -64,6 +64,7 @@ func NewL2Genesis(config *DeployConfig, block *types.Block) (*core.Genesis, erro
 		// [Kroma: START]
 		// kroma-geth is not yet ready. it should be upstream to v1.101304.2
 		// InteropTime:                   config.InteropTime(block.Time()),
+		BurgundyTime: config.BurgundyTime(block.Time()),
 		//
```
Comment (seolaoh) [inline kroma-chain-ops/genesis/genesis.go:68]:
```suggestion
```

File: ops-devnet/docker-compose.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ services:
 
   l2:
     pid: host # allow debugging
-    image: kromanetwork/geth:v0.4.4
+    image: kromanetwork/geth:tge
```
Comment (seolaoh) [inline ops-devnet/docker-compose.yml:34]: Why did you change it?

File: go.mod [hunk 3]
```diff
@@ -198,4 +198,4 @@ require (
 
 replace github.com/ethereum-optimism/optimism v1.4.0 => ./
 
-replace github.com/ethereum/go-ethereum v1.13.8 => github.com/kroma-network/go-ethereum v0.4.4
+replace github.com/ethereum/go-ethereum v1.13.8 => github.com/kroma-network/go-ethereum v0.4.3-0.20240304033547-c8ef45c4754e
```
Comment (seolaoh) [inline go.mod:201]: Why did you change it?

File: op-e2e/e2eutils/setup.go [hunk 4]
```diff
@@ -24,7 +24,7 @@ var testingJWTSecret = [32]byte{123}
 func WriteDefaultJWT(t TestingBase) string {
 	// Sadly the geth node config cannot load JWT secret from memory, it has to be a file
 	jwtPath := path.Join(t.TempDir(), "jwt_secret")
-	if err := os.WriteFile(jwtPath, []byte(hexutil.Encode(testingJWTSecret[:])), 0600); err != nil {
+	if err := os.WriteFile(jwtPath, []byte(hexutil.Encode(testingJWTSecret[:])), 0o600); err != nil {
```
Comment (0xHansLee) [inline op-e2e/e2eutils/setup.go:27]: Why do you make this change?

**Review by 0xHansLee — COMMENTED** [review]
2 things I wanna say:

1. Do we need override flag for Burgundy? If so, it should be added to `op-node/flags/flags.go`.
2. Why don't we add `BurgundyTime` to rollup config for both mainnet and sepolia as nil in `op-node/chaincfg/chains.go`?

File: ops-devnet/docker-compose.yml (hunk 2 above)
Comment (Pangssu) [reply ops-devnet/docker-compose.yml:34]: This PR requires a version of geth with the burgundy changes.

File: go.mod (hunk 3 above)
Comment (Pangssu) [reply go.mod:201]: This PR requires a version of geth with the burgundy changes.

File: op-e2e/e2eutils/setup.go (hunk 4 above)
Comment (Pangssu) [reply op-e2e/e2eutils/setup.go:27]: IntelliJ changed it automatically, and I couldn't find a way to not change it.

File: ops-devnet/docker-compose.yml (hunk 2 above)
Comment (Pangssu) [reply ops-devnet/docker-compose.yml:34]: FYI, this should change when geth is released.

File: op-e2e/e2eutils/setup.go (hunk 4 above)
Comment (0xHansLee) [reply op-e2e/e2eutils/setup.go:27]: Got it

**Comment by Pangssu** [comment]
Closing this PR because our team have made an internal decision to not create `MintTokenTx`, and the network upgrade is no longer needed.

**Pull request closed without merging**